scripts/registry.js

```diff
@@ -1,88 +1,94 @@
 
   static async create(workdir) {
-    await copy(path.resolve(__dirname, '../etc/verdaccio-config.yaml'), path.resolve(workdir, 'verdaccio.yaml'));
+    mkdirSync(workdir, { recursive: true });
 
-    const proc = spawnNpx(['--yes', 'verdaccio', '-c', 'verdaccio.yaml'], {
-      cwd: workdir,
-      stdio: 'inherit',
-      shell,
-      detached: true,
-    });
-    return new this(proc, workdir);
-  }
+    const app = await runServer({
+      self_path: workdir,
+
+      // ...config,
+      storage: path.resolve(workdir, 'storage'),
+
+      web: {
+        title: 'Verdaccio',
+      },
 
-  async ready() {
-    const logPath = path.resolve(this.workdir, 'verdaccio.log');
-    await untilExists(logPath, 120);
-    const tail = new Tail(logPath, {
-      fromBeginning: true,
+      auth: {
+        htpasswd: {
+          file: path.resolve(workdir, 'htpasswd'),
+        },
+      },
+
+      uplinks: {
+        npmjs: {
+          url: 'https://registry.npmjs.org/',
+        },
+      },
+
+      packages: {
+        // Note that the Temporal packages don't proxy npmjs to ensure we test the correct packages
+        '@temporalio/*': {
+          access: '$all',
+          publish: '$all',
+          unpublish: '$all',
+        },
+        temporalio: {
+          access: '$all',
+          publish: '$all',
+          unpublish: '$all',
+        },
+        '@*/*': {
+          access: '$all',
+          publish: '$all',
+          unpublish: '$all',
+          proxy: 'npmjs',
+        },
+        '**': {
+          access: '$all',
+          publish: '$all',
+          unpublish: '$all',
+          proxy: 'npmjs',
+        },
+      },
+
+      server: {
+        keepAliveTimeout: 60,
+      },
+      // We have some really large packages (e.g. worker)
+      max_body_size: '200mb',
+
+      logs: {
+        type: 'file',
+        format: 'json',
+        path: path.resolve(workdir, 'verdaccio.log'),
+        level: 'info',
+      },
     });
-    try {
-      await new Promise((resolve, reject) => {
-        tail.on('line', (line) => {
-          const parsed = JSON.parse(line);
-          if (parsed.addr) {
-            resolve();
-          }
-        });
-
-        tail.on('error', reject);
-        setTimeout(async () => {
-          let contents;
-          try {
-            contents = await readFile(logPath, 'utf8');
-            // Sometimes (mostly in Windows) tail can miss updates.
-            // Use this workaround as a last resort to recover and avoid failing CI.
-            const found = contents
-              .split('\n')
-              .map(JSON.parse)
-              .find((parsed) => parsed.addr);
-            if (found) {
-              resolve();
-            }
-          } catch (e) {
-            contents = `Error ${e}`;
-          }
-          reject(new Error(`Timed out waiting for verdaccio - ${contents}`));
-        }, 60 * 1000);
-      });
-    } finally {
-      tail.unwatch();
-    }
+
+    app.listen(4873);
```

Comment:
Is there a way to wait for the server to listen?

Reply:
I think this should do it. `app` is actually an instance of nodejs' `http`.

```
await new Promise((resolve, reject) => {
    try {
        app.listen(4873, resolve)
    } catch (e) {
        reject(e)
    }
})
```

Reply:
Yeah, that's what I usually do, you might need to add `app.on('error', reject)`.